### PR TITLE
Admin can store session in database

### LINKED ISSUE #73: Store sessions in database
Save information about the upcoming sessions and render those on the home page.
Information about sessions can be added/updated using the `Admin` interface.

### PR DESCRIPTION
**Pull request by vaibhavsingh97**
**Status**:
- [x] Ready
- [ ] In Development
- [ ] On Hold

**Which issue does this PR fix?**: 
fixes #73 

This PR will allow admin to add and delete sessions. Also, Latest 3 sessions will be displayed on the homepage.

**Why do we need this PR?**:
Currently, We have to add sessions manually and to do that we have to edit page itself. Now, admin can add the session in one place and changes will be reflected to the homepage.

**Demo (optional)**:
Admin Screens:
image
image

User screen:
image

**Testing instructions:**:
1. `git checkout develop`
2. `python manage.py makemigrations`
3. `python manage.py migrate`
4. `python manage.py collectstatic`
5. `python manage.py runserver`
6.  To add sessions, goto `http://127.0.0.1:8000/admin/main/chatsession/add/`

### CODE REVIEW

**Comment by vaibhavsingh97** [comment]
@tapasweni-pathak @nikhita @jarifibrahim Please review

**Comment by jarifibrahim** [comment]
@tapasweni-pathak Please enable Travis builds for Pull Requests.

File: oshc/main/models.py [hunk 1]
```diff
@@ -2,5 +2,28 @@
 from __future__ import unicode_literals
 
 from django.db import models
+from django.forms import ModelForm, Textarea
 
 # Create your models here.
```
Comment (jarifibrahim) [inline oshc/main/models.py]: Please remove this comment

File: oshc/main/models.py [hunk 2]
```diff
@@ -2,5 +2,28 @@
...
+
+
+class chatSession(models.Model):
+    title = models.CharField(max_length=128, help_text="Require title of the session")
+    profile_name = models.CharField(max_length=128, help_text="Require name of the person")
+    profile_url = models.URLField(help_text="Require url of the person")
+    description = models.CharField(max_length=512, help_text="Require description of the session")
```
Comment (jarifibrahim) [inline oshc/main/models.py]: `models.TextField` would be better for description. We'll have `Textarea` widget by default.

File: oshc/main/models.py [hunk 3]
```diff
@@ -2,5 +2,28 @@
 from __future__ import unicode_literals
 
 from django.db import models
+from django.forms import ModelForm, Textarea
 
 # Create your models here.
+
+
+class chatSession(models.Model):
+    title = models.CharField(max_length=128, help_text="Require title of the session")
```
Comment (jarifibrahim) [inline oshc/main/models.py]: `help_text="Title of the session"` would be more appropriate. All the fields are mandatory by default. You don't need to add an explicit `require` :)

The same goes for all the fields in the model.

File: oshc/main/models.py [hunk 4]
```diff
@@ -2,5 +2,28 @@
...
+    description = models.CharField(max_length=512, help_text="Require description of the session")
+    start_date = models.DateTimeField()
+    end_date = models.DateTimeField()
+    register_url = models.URLField(help_text="Require url for the registration of event")
+
+    def __str__(self):
+        return self.title
+
+
+class PostModelForm(ModelForm):
```
Comment (jarifibrahim) [inline oshc/main/models.py]: Why do we need `PostModelForm`?

File: oshc/main/models.py [hunk 5]
```diff
@@ -2,5 +2,28 @@
...
+    class Meta:
+        model = chatSession
+        fields = ('title', 'profile_name', 'profile_url', 'description', 'start_date', 'end_date', 'register_url')
```
Comment (jarifibrahim) [inline oshc/main/models.py]: All the model fields will be displayed on the admin page by default. You don't need to add them explicitly.

File: oshc/main/models.py [hunk 6]
```diff
@@ -2,5 +2,28 @@
...
+        widgets = {
+            'description': Textarea(attrs={'cols': 50, 'rows': 6}),
```
Comment (jarifibrahim) [inline oshc/main/models.py]: If we use `TextField` for description, this won't be necessary. `Textarea` is the default widget for `TextField`

File: oshc/main/templates/index.html [hunk 7]
```diff
@@ -1,63 +1,67 @@
...
+          </div>
+          <div class="panel-body">
+            <a href="{{ session.profile_url }}">{{ session.profile_name }}</a> {{ session.description }}
+            <div class="line-gap-10px"></div>
+            <i>{{ session.start_date }} – {{ session.end_date }}</i>
+            <div class="line-gap-10px"></div>
+            <button type="button" class="btn btn-lg btn-primary sRegister">
+                                        <a href="{{ session.register_url }}">Register
+                                            Now!</a>
+                                    </button>
```
Comment (jarifibrahim) [inline oshc/main/templates/index.html]: Fix indentation.

File: oshc/main/admin.py [hunk 8]
```diff
@@ -1,6 +1,15 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
+from main.models import chatSession, PostModelForm
 
 from django.contrib import admin
 
 # Register your models here.
```
Comment (jarifibrahim) [inline oshc/main/admin.py]: Please remove this comment.

File: oshc/main/admin.py [hunk 9]
```diff
@@ -1,6 +1,15 @@
...
+
+
+class chatSessionAdmin(admin.ModelAdmin):
+    list_display = ('title', 'start_date')
+    form = PostModelForm
```
Comment (jarifibrahim) [inline oshc/main/admin.py]: We don't need `PostModelForm`.

**Review by jarifibrahim — CHANGES_REQUESTED** [review]
@vaibhavsingh97 Could you please add a few tests for this PR?

**Comment by vaibhavsingh97** [comment]
@jarifibrahim Please review again

**Comment by tapaswenipathak** [comment]
@jarifibrahim It is already running on the pull requests.

**Comment by vaibhavsingh97** [comment]
Closing this PR as lot of issues.

**Pull request closed without merging**